openapi-core/src/main/java/io/ballerina/openapi/core/generators/schema/TypeGeneratorUtils.java

```diff
@@ -276,12 +283,20 @@ private static RecordFieldWithDefaultValueNode getRecordFieldWithDefaultValueNod
      * @param fieldSchema Schema for data type
      * @return {@link MetadataNode}
      */
-    public static AnnotationNode generateConstraintNode(String typeName, Schema<?> fieldSchema) {
+    public static AnnotationNode generateConstraintNode(String typeName, Schema<?> fieldSchema)
+            throws BallerinaOpenApiException {
         if (isConstraintAllowed(typeName, fieldSchema)) {
-            if (GeneratorUtils.isStringSchema(fieldSchema)) {
+            String ballerinaType = convertOpenAPITypeToBallerina(fieldSchema);
+            //The current tool still supports byte and binary formats in string types.
+            //The rest of the string formats are mapped to the string type. Therefore, we do not provide constraint
+            //support for byte and binary formats.
+            //TODO: Please verify if constraints are supported when attempting to add new string format type support.
+            if ((GeneratorUtils.isStringSchema(fieldSchema) && ballerinaType.equals(STRING))) {
```

Comment:
Do we still need to check if its a string schema? it seems redundant to me. @aneeshafedo WDYT?

Reply:
No we don't. `ballerinaType.equals(STRING)` would be enough.

Reply:
During the call, we discussed the addition of these checks, as `isStringSchema` only verifies the type of the schema within it and doesn't capture the format details. Let me check if we can eliminate the schema check and retain the remaining Ballerina type.

Reply:
Updated here https://github.com/ballerina-platform/openapi-tools/pull/1508/commits/59db7ee7bb123c4db7d1fb4d5e4b9a83ee12c240